src/SSHDebugPS/Docker/DockerContainerInstance.cs

```diff
@@ -2,31 +2,41 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Collections.Generic;
 using System.Diagnostics;
+
+using Microsoft.DebugEngineHost;
+using Microsoft.SSHDebugPS.Utilities;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.SSHDebugPS.Docker
 {
     public class DockerContainerInstance : ContainerInstance
     {
+        private static readonly string Property_ExceptionName = "ExceptionName";
         /// <summary>
         /// Create a DockerContainerInstance from the results of docker ps in JSON format
         /// </summary>
-        public static DockerContainerInstance Create(string json)
+        public static bool TryCreate(string json, out DockerContainerInstance instance)
         {
+            instance = null;
             try
             {
                 JObject obj = JObject.Parse(json);
-                var instance = obj.ToObject<DockerContainerInstance>();
-                if (instance != null)
-                    return instance;
+                instance = obj.ToObject<DockerContainerInstance>();
             }
             catch (Exception e)
             {
-                Debug.Fail(e.ToString());
+                List<KeyValuePair<string, object>> eventProperties = new List<KeyValuePair<string, object>>();
+                eventProperties.Add(new KeyValuePair<string, object>(Property_ExceptionName, e.GetType().Name));
+                HostTelemetry.SendEvent(Telemetry.Event_DockerPSParseFailure, eventProperties.ToArray());
```

Comment:
why create a list to add an item and then call toArray on it ? can you define an array with one keyvalue pair inline?